src/platform/ESP32/ConfigurationManagerImpl.cpp

```diff
@@ -102,9 +102,28 @@ CHIP_ERROR ConfigurationManagerImpl::_Init()
 
 CHIP_ERROR ConfigurationManagerImpl::_GetPrimaryWiFiMACAddress(uint8_t * buf)
 {
-    return esp_wifi_get_mac(WIFI_IF_STA, buf);
+    wifi_mode_t mode;
+    esp_wifi_get_mode(&mode);
+    if ((mode == WIFI_MODE_AP) || (mode == WIFI_MODE_APSTA))
+        return MapConfigError(esp_wifi_get_mac(WIFI_IF_AP, buf));
+    else
+        return MapConfigError(esp_wifi_get_mac(WIFI_IF_STA, buf));
+}
+CHIP_ERROR ConfigurationManagerImpl::MapConfigError(esp_err_t error)
+{
+    switch (error)
+    {
+    case ESP_OK:
+        return CHIP_NO_ERROR;
+    case ESP_ERR_WIFI_NOT_INIT:
+        return CHIP_ERROR_WELL_UNINITIALIZED;
+    case ESP_ERR_INVALID_ARG:
+    case ESP_ERR_WIFI_IF:
+        return CHIP_ERROR_INVALID_ARGUMENT;
+    default:
+        return CHIP_ERROR_NOT_IMPLEMENTED;
```

Comment:
maybe INTERNAL would be more accurate than 'Not implemented'

Reply:
Fixed